### DISCUSSION TITLE
Pairing and Bonding

### DISCUSSION THREAD

**Discussion by sisaacks** [dotnet-bluetooth-le/dotnet-bluetooth-le#684] [Q&A] [answered]
I have one device where I have a requirement to pair and bond with a key.  Is there a way I can do this with Plugin.BLE?  I have searched and searched and searched but I cannot seem to find anything :(

**Comment by janusw** [comment]
There is an open PR for basic bonding support on Android, but it's still work in progress: #691
I hope it will be available in a release soon.

**Reply by janusw** [reply, marked as answer]
PR #691 has been merged in the meantime, and release 3.0.0 (which includes this feature) has just been released.

**Comment by AskBojesen** [comment]
We are also looking into bonding, but we do not know yet how we want to do it.

It is a complex situation - you might want to support a subset of the pairing and bonding in this plugin - case by case....

https://academy.nordicsemi.com/courses/bluetooth-low-energy-fundamentals/lessons/lesson-5-bluetooth-le-security-fundamentals/topic/pairing-process/

**Reply by axa88** [reply]
> We are also looking into bo nding, but we do not know yet how we want to do it.
> 
> It is a complex situation

Yea its a bit complex, mostly due to consideration of implementing it in a way that keeps everything familiar across platforms.
But i need this on windows so do it now.

And I imagine this discussion and effort only addressed Android? see #730.
So it should be asked, what other, if any, additional features should be attempted for Android?
Noticed talk of [programmatically storing/using a passkey](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/pull/691#issuecomment-1666950254), seems possible with compromises, but is it worth the effort?  Need input.

Not set up for iOS at the moment.

**Reply by axa88** [reply]
> Noticed talk of [programmatically storing/using a passkey](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/pull/691#issuecomment-1666950254), seems possible with compromises, but is it worth the effort? Need input.

This is not possible.  While the OS triggered pairing UI can be automatically dismissed with a side effect of a native `Java.Lang.SecurityException` exception thrown, the pair cannot be programmatically accepted without the `BLUETOOTH PRIVILEGED` permission which is not available to 3rd party apps.

Im not aware of any other pairing improvements that can be made to the Android implementation

**Reply by axa88** [reply]
> Yea its a bit complex, mostly due to consideration of implementing it in a way that keeps everything familiar across platforms.
> But i need this on windows so do it now.

Now for windows #906
